.gitlab-ci.yml

```diff
@@ -1,118 +1,142 @@
+image: stackstate/stackstate-agent-runner-circle:omnibus-cache
+
 stages:
-- tests
-- build
+- prepare
+- build_and_test
 - publish
 
-unit_tests:
- stage: tests
- image: stackstate/stackstate-agent-runner-circle:latest
- before_script:
+before_script:
  - mkdir -p /go/src/github.com/StackVista
  - ln -s $CI_PROJECT_DIR  /go/src/github.com/StackVista/stackstate-agent
- script:
+   # Either we already have a venv, or we still need to make it (as part of get_deps.sh)
+ - source $CI_PROJECT_DIR/venv/bin/activate || source .gitlab-scripts/ensure_deps.sh
  - cd /go/src/github.com/StackVista/stackstate-agent
- - pip install -r requirements.txt
- - inv deps
- - inv -e agent.build --race --precompile-only
- - inv -e test --coverage --race --profile --fail-on-fmt --cpus 4 --build-exclude clusterchecks,kubeapiserver,kubelet
+
+dependencies_clear_cache:
+  stage: prepare
+  cache:
+    policy: push
+    key: "agent-deps"
+    paths:
+      - vendor
+      - venv
+  when: manual
+  script:
+    - rm -rf $CI_PROJECT_DIR/venv
+    - rm -rf $CI_PROJECT_DIR/vendor
+
+deps:
+  stage: prepare
+  cache:
+    key: "agent-deps"
+    paths:
+      - vendor
+      - venv
+  script:
+    - pip install -r requirements.txt
+    - inv deps
+  artifacts:
+    paths:
+        - $CI_PROJECT_DIR/vendor
+        - $CI_PROJECT_DIR/venv
+    expire_in: 1 week
+
+unit_tests:
+  stage: build_and_test
+  dependencies:
+    - deps
+  script:
+    #We run another inv deps here to get the go dependencies
+    - inv deps
```

Comment:
wait the first `inv deps` does already get the go dependencies, no ?

Reply:
that one is conditional